src/silx/io/spech5.py

```diff
@@ -278,8 +278,12 @@ def _parse_UB_matrix(header_line):
 
     :param str header_line: G3 header line
     :return: UB matrix
+    :raises ValueError: For malformed UB matrix header line
     """
-    return numpy.array(list(map(float, header_line.split()))).reshape((1, 3, 3))
+    values = list(map(float, header_line.split()))  # Can raise ValueError
+    if len(values) < 9:
```

Comment:
well I guess this should be if len(values) != 9 no ?

Reply:
but could bring to some "errors that might be ignore" as some comments...

Reply:
I agree it would be best, but the previous behavior was to provide something if `len(values) >= 9` and I have no clue how this is used/abused (and if it was on purpose or not), so I prefer to keep backward compatibility.